include/libtorrent/aux_/hasher512.hpp

```diff
@@ -122,7 +122,7 @@ namespace aux {
 #elif TORRENT_USE_CRYPTOAPI_SHA_512
 		aux::crypt_hash<CALG_SHA_512, PROV_RSA_AES> m_context;
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_CTX m_context;
+		EVP_MD_CTX *m_context;
```

Comment:
It seems openssl has intentionally made these structures opaque as it forward declares it...
https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes#Compatibility_Layer

Reply:
@arvidn 
This changes the structure size so it breaks ABI compatibility right?
Should I retarget to master branch instead?

Reply:
yes, ABI changes need to go into `master`. Alternatively, it could be a new build configuration with its own inline namespace, separating it from previous symbols. But I suspect that's overkill. The lead time for this API to go away is probably long